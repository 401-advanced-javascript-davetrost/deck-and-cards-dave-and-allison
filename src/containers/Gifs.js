import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeckOfComps from './DeckOfComps';
import Card from '../components/Card';
import { getGifs } from '../services/hey-arnold-api';

export default class Gifs extends Component {
  state = {
    gifs: []
  }

  componentDidMount() {
    getGifs({ count: this.props.count })
      .then(gifs => {
        this.setState({ gifs });
      });
  }

  render() {
    return (
      <DeckOfComps
        title="GIFs"
        items={this.state.gifs.map(gif => ({ 
          ...gif, 
          title: gif._id
        }))}
        Component={Card}
      />
    );
  }
  
  static propTypes = {
    count: PropTypes.number
  }


}

