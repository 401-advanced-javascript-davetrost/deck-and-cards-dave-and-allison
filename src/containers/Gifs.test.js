import React from 'react';
import { shallow } from 'enzyme';
import Gifs from './Gifs';

jest.mock('../services/hey-arnold-api', () => ({
  getGifs() {
    return Promise.resolve([
      {
        '_id': '5da23754845fd2cb76d59ef2',
        'image': 'https://media.tenor.com/images/e7623921965651fa6fba7e0c37892044/tenor.gif'
      }
    ]);
  }
}));


describe('Gifs component', () => {

  it('renders', () => {
    const wrapper = shallow(
      <Gifs count={5} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
