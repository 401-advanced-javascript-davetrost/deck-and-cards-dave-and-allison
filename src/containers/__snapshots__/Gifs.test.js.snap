// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifs component renders 1`] = `
<DeckOfComps
  Component={[Function]}
  items={Array []}
  title="GIFs"
/>
`;
